basetest.pm

```diff
@@ -705,20 +705,24 @@ sub parse_serial_output_qemu {
             my $type    = $regexp_table->{type};
 
             # Input parameters validation
-            die "Wrong type defined for serial failure. Only 'soft' or 'hard' allowed. Got: $type" if $type !~ /^soft|hard|fatal$/;
+            die "Wrong type defined for serial failure. Only 'info', 'soft', 'hard' or 'fatal' allowed. Got: $type" if $type !~ /^info|soft|hard|fatal$/;
             die "Message not defined for serial failure for the pattern: '$regexp', type: $type"   if !defined $message;
 
             # If you want to match a simple string please be sure that you create it with quotemeta
             if (!exists $regexp_matched{$regexp} and $line =~ /$regexp/) {
                 $regexp_matched{$regexp} = 1;
-                my $fail_type = 'softfail';
-                if ($type =~ 'hard|fatal') {
-                    $die                   = 1;
-                    $fail_type             = 'fail';
-                    $self->{fatal_failure} = $type eq 'fatal';
+                if ($type eq 'info') {
+                    testapi::record_info("$type", "$message");
+                } else {
+                    my $fail_type = 'softfail';
+                    if ($type =~ 'hard|fatal') {
+                        $die                   = 1;
+                        $fail_type             = 'fail';
+                        $self->{fatal_failure} = $type eq 'fatal';
+                    }
+                    $self->record_resultfile($message, $message . " - Serial error: $line", result => $fail_type);
+                    $self->{result} = $fail_type;
```

Comment:
you can still use `$self->record_resultfile` here so no need to make it that different and hence can avoid the higher complexity of multiple levels of if.

```suggestion
                my $fail_type = 'softfail';
                if ($type eq 'info') {
                    $fail_type = 'ok';
                }
                elsif ($type =~ 'hard|fatal') {
                    $die                   = 1;
                    $fail_type             = 'fail';
                    $self->{fatal_failure} = $type eq 'fatal';
                }
                $self->record_resultfile($message, $message . " - Serial error: $line", result => $fail_type);
                $self->{result} = $fail_type;
```

Reply:
To answer your question where "record_info" is to be used: You don't need to call it directly. Sorry if the ticket was phrased not entirely ambiguously in that regard. Actually I think it's a good practice to not call any public test API internally in other os-autoinst modules (following "design by contract" principles).